### ISSUE TITLE
pip-compile fails on mac when scipy is in requirements.in

### ISSUE THREAD

**Issue by RussTedrake** [jazzband/pip-tools#1660]
It's hard for me to tell if this is an issue with `pip-compile` or with `scipy`, but the error occurs when running `pip-compile`, so I hope it's helpful to report it here.

#### Environment Versions

1. OS Type: Mac OS 11.6.2
1. Python version: 3.9.13
1. pip version: 22.1.1
1. pip-tools version: 6.8.0

#### Steps to replicate

```
% cat test.in
scipy
% pip-compile --rebuild test.in
```

#### Expected result

Successful generation of the requirements file.

#### Actual result

```
% pip-compile --rebuild test.in
    error: subprocess-exited-with-error
    
    × Getting requirements to build wheel did not run successfully.
    │ exit code: 1
    ╰─> [59 lines of output]
        The Meson build system
        Version: 0.62.2
        Source dir: /private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy
        Build dir: /private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy/.mesonpy-6akzkm97/build
        Build type: native build
        Project name: SciPy
        Project version: 1.9.0
        C compiler for the host machine: cc (clang 13.0.0 "Apple clang version 13.0.0 (clang-1300.0.29.30)")
        C linker for the host machine: cc ld64 711
        C++ compiler for the host machine: c++ (clang 13.0.0 "Apple clang version 13.0.0 (clang-1300.0.29.30)")
        C++ linker for the host machine: c++ ld64 711
        Host machine cpu family: aarch64
        Host machine cpu: arm64
        Compiler for C supports arguments -Wno-unused-but-set-variable: NO
        Library m found: YES
        Fortran compiler for the host machine: gfortran (gcc 11.3.0 "GNU Fortran (Homebrew GCC 11.3.0_2) 11.3.0")
        Fortran linker for the host machine: gfortran ld64 711
        Program cython found: YES (/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/bin/cython)
        Program pythran found: YES (/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/bin/pythran)
        Program cp found: YES (/bin/cp)
        Program python3 found: YES (/opt/homebrew/Cellar/python@3.9/3.9.13_1/bin/python3.9)
        Found pkg-config: /opt/homebrew/bin/pkg-config (0.29.2)
        Library npymath found: YES
        Library npyrandom found: YES
        Found CMake: /opt/homebrew/bin/cmake (3.23.3)
        Run-time dependency openblas found: NO (tried pkgconfig, framework and cmake)
        
        ../../scipy/meson.build:120:0: ERROR: Dependency "openblas" not found, tried pkgconfig, framework and cmake
        
        A full log can be found at /private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy/.mesonpy-6akzkm97/build/meson-logs/meson-log.txt
        + meson setup --native-file=/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy/.mesonpy-native-file.ini -Ddebug=false -Doptimization=2 --prefix=/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9 /private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy /private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy/.mesonpy-6akzkm97/build
        Traceback (most recent call last):
          File "/opt/homebrew/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
            main()
          File "/opt/homebrew/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
            json_out['return_val'] = hook(**hook_input['kwargs'])
          File "/opt/homebrew/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
            return hook(config_settings)
          File "/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/lib/python3.9/site-packages/mesonpy/__init__.py", line 909, in get_requires_for_build_wheel
            with _project(config_settings) as project:
          File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/contextlib.py", line 119, in __enter__
            return next(self.gen)
          File "/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/lib/python3.9/site-packages/mesonpy/__init__.py", line 888, in _project
            with Project.with_temp_working_dir(
          File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/contextlib.py", line 119, in __enter__
            return next(self.gen)
          File "/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/lib/python3.9/site-packages/mesonpy/__init__.py", line 547, in with_temp_working_dir
            yield cls(source_dir, tmpdir, build_dir)
          File "/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/lib/python3.9/site-packages/mesonpy/__init__.py", line 463, in __init__
            self._configure(reconfigure=bool(build_dir) and not native_file_mismatch)
          File "/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/lib/python3.9/site-packages/mesonpy/__init__.py", line 494, in _configure
            self._meson(
          File "/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/lib/python3.9/site-packages/mesonpy/__init__.py", line 477, in _meson
            return self._proc('meson', *args)
          File "/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-build-env-mvxd2b9r/overlay/lib/python3.9/site-packages/mesonpy/__init__.py", line 472, in _proc
            subprocess.check_call(list(args))
          File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
            raise CalledProcessError(retcode, cmd)
        subprocess.CalledProcessError: Command '['meson', 'setup', '--native-file=/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy/.mesonpy-native-file.ini', '-Ddebug=false', '-Doptimization=2', '--prefix=/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9', '/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy', '/private/var/folders/6p/t1z4stds49974ltdtn62_gzw0000gr/T/pip-resolver-gaswgbs0/scipy/.mesonpy-6akzkm97/build']' returned non-zero exit status 1.
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
Traceback (most recent call last):
  File "/opt/homebrew/bin/pip-compile", line 8, in <module>
    sys.exit(cli())
  File "/opt/homebrew/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/piptools/scripts/compile.py", line 487, in cli
    results = resolver.resolve(max_rounds=max_rounds)
  File "/opt/homebrew/lib/python3.9/site-packages/piptools/resolver.py", line 266, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "/opt/homebrew/lib/python3.9/site-packages/piptools/resolver.py", line 356, in _resolve_one_round
    their_constraints.extend(self._iter_dependencies(best_match))
  File "/opt/homebrew/lib/python3.9/site-packages/piptools/resolver.py", line 469, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "/opt/homebrew/lib/python3.9/site-packages/piptools/repositories/pypi.py", line 250, in get_dependencies
    self._dependencies_cache[ireq] = self.resolve_reqs(
  File "/opt/homebrew/lib/python3.9/site-packages/piptools/repositories/pypi.py", line 213, in resolve_reqs
    results = resolver._resolve_one(reqset, ireq)
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/resolution/legacy/resolver.py", line 500, in _resolve_one
    dist = self._get_dist_for(req_to_install)
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/resolution/legacy/resolver.py", line 453, in _get_dist_for
    dist = self.preparer.prepare_linked_requirement(req)
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 428, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 497, in _prepare_linked_requirement
    dist = _get_prepared_distribution(
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 58, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/distributions/sdist.py", line 48, in prepare_distribution_metadata
    self._install_build_reqs(finder)
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/distributions/sdist.py", line 118, in _install_build_reqs
    build_reqs = self._get_build_requires_wheel()
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/distributions/sdist.py", line 95, in _get_build_requires_wheel
    return backend.get_requires_for_build_wheel()
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/utils/misc.py", line 685, in get_requires_for_build_wheel
    return super().get_requires_for_build_wheel(config_settings=cs)
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 172, in get_requires_for_build_wheel
    return self._call_hook('get_requires_for_build_wheel', {
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 322, in _call_hook
    self._subprocess_runner(
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/utils/subprocess.py", line 252, in runner
    call_subprocess(
  File "/opt/homebrew/lib/python3.9/site-packages/pip/_internal/utils/subprocess.py", line 224, in call_subprocess
    raise error
pip._internal.exceptions.InstallationSubprocessError: Getting requirements to build wheel exited with 1
```

Pegging the scipy revision did not help in `test.in` did not help.

I confirmed that `openblas` is installed (via homebrew) and update to date.

Perhaps notably, `scipy` is already installed on this system (via homebrew).  
```
% pip3 install scipy
Requirement already satisfied: scipy in /opt/homebrew/lib/python3.9/site-packages (1.9.0)
Requirement already satisfied: numpy<1.25.0,>=1.18.5 in /opt/homebrew/lib/python3.9/site-packages (from scipy) (1.23.0)
```

**Comment by atugushev** [comment]
This might be an error while building with `meson`:
```
../../scipy/meson.build:120:0: ERROR: Dependency "openblas" not found, tried pkgconfig, framework and cmake
```

Doesn't look like an issue with pip-tools.

**Comment by eli-schwartz** [comment]
```
        Found pkg-config: /opt/homebrew/bin/pkg-config (0.29.2)
```

Does this pkg-config program find openblas installed when running `pkg-config --modversion openblas`? The --debug flag may also provide useful information. Since the pkg-config binary in question comes from homebrew, I would expect that it also is configured to find pkg-config files installed by homebrew packages...

(The openblas package should install e.g. /opt/homebrew/lib/pkgconfig/openblas.pc)

**Comment by RussTedrake** [comment]
Thanks all.  I don't have `/opt/homebrew/lib/pkgconfig/openblas.pc` and (as expected) pkg-config fails:
```
% pkg-config --modversion openblas
Package openblas was not found in the pkg-config search path.
Perhaps you should add the directory containing `openblas.pc'
to the PKG_CONFIG_PATH environment variable
No package 'openblas' found
```
As a sanity check, I did `homebrew reinstall openblas`, but still no `openblas.pc`.
I do have `/opt/homebrew/Cellar/openblas/0.3.20/lib/pkgconfig/openblas.pc`, but it seems it was not symlinked.

```
ln -s /opt/homebrew/Cellar/openblas/0.3.20/lib/pkgconfig/openblas.pc /opt/homebrew/lib/pkgconfig/openblas.pc
```
resolved the problem for me. (`pip-compile` now succeeds).

I guess this is an issue with the homebrew formula?

**Comment by eli-schwartz** [comment]
I would definitely open a homebrew formula issue.

**Comment by RussTedrake** [comment]
As I was running brew doctor, etc, to prepare for the report, I saw this:
```
% brew unlink openblas && brew link openblas
Unlinking /opt/homebrew/Cellar/openblas/0.3.20... 1 symlinks removed.
Warning: openblas is keg-only and must be linked with `--force`
```
I've verified that `brew link --force openblas` would have generated the symlink for me; which is slightly less manual.  So it seems that the problem is more that `scipy` assumes that there should be a pkgconfig for openblas, but homebrew says you must `--force` to get it installed.

At any rate, I hope this helps the next person who stumbles on it.  Thanks again!
